plugins/workload-management/src/test/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceServiceTests.java

```diff
@@ -298,4 +304,177 @@ public void testMaxQueryGroupCount() {
         queryGroupPersistenceService.setMaxQueryGroupCount(50);
         assertEquals(50, queryGroupPersistenceService.getMaxQueryGroupCount());
     }
+
+    /**
+     * Tests updating a QueryGroup with all fields
+     */
+    public void testUpdateQueryGroupAllFields() {
+        QueryGroup updated = builder().name(NAME_ONE)
+            ._id(_ID_ONE)
+            .mode("enforced")
+            .resourceLimits(Map.of(ResourceType.fromName(MEMORY_STRING), 0.15))
+            .updatedAt(1690934400000L)
+            .build();
+        UpdateQueryGroupRequest updateQueryGroupRequest = new UpdateQueryGroupRequest(
+            NAME_ONE,
+            ResiliencyMode.fromName("enforced"),
+            Map.of(ResourceType.fromName(MEMORY_STRING), 0.15),
+            1690934400000L
+        );
+        ClusterState newClusterState = queryGroupPersistenceService().updateQueryGroupInClusterState(
+            updateQueryGroupRequest,
+            clusterState()
+        );
+        List<QueryGroup> updatedQueryGroups = new ArrayList<>(newClusterState.getMetadata().queryGroups().values());
+        assertEquals(2, updatedQueryGroups.size());
+        List<QueryGroup> expectedList = new ArrayList<>();
+        expectedList.add(queryGroupTwo);
+        expectedList.add(updated);
+        assertEqualQueryGroups(expectedList, updatedQueryGroups);
+    }
+
+    /**
+     * Tests updating a QueryGroup with only updated resourceLimits
+     */
+    public void testUpdateQueryGroupResourceLimitsOnly() {
+        QueryGroup updated = builder().name(NAME_ONE)
+            ._id(_ID_ONE)
+            .mode(MONITOR_STRING)
+            .resourceLimits(Map.of(ResourceType.fromName(MEMORY_STRING), 0.15))
+            .updatedAt(1690934400000L)
+            .build();
+        UpdateQueryGroupRequest updateQueryGroupRequest = new UpdateQueryGroupRequest(
+            NAME_ONE,
+            ResiliencyMode.fromName(MONITOR_STRING),
+            Map.of(ResourceType.fromName(MEMORY_STRING), 0.15),
```

Comment:
This updated value should be different from the existing one to test the success scenario

Reply:
The resource limit is different. Previously the state has queryGroupOne, which has resource limit as `Map.of(fromName(MEMORY_STRING), 0.3)`